src/buildblock/ML_norm.cxx

```diff
@@ -1668,6 +1839,40 @@ double KL(const FanProjData& d1, const FanProjData& d2, const double threshold)
   return static_cast<double>(sum);
 }
 */
-                   
+
+template void make_det_pair_data_help
+(DetPairData&, const ProjDataInfoBlocksOnCylindricalNoArcCorr&, const ProjData&, const int, const int);
+template void make_det_pair_data_help
+(DetPairData&, const ProjDataInfoCylindricalNoArcCorr&, const ProjData&, const int, const int);
+
+template void make_det_pair_data_help
+(DetPairData&, const ProjDataInfoBlocksOnCylindricalNoArcCorr&, const int, const int);
+template void make_det_pair_data_help
+(DetPairData&, const ProjDataInfoCylindricalNoArcCorr&, const int, const int);
+
+template void set_det_pair_data_help
+(ProjData&, const ProjDataInfoBlocksOnCylindricalNoArcCorr, const DetPairData&, const int, const int);
+template void set_det_pair_data_help
+(ProjData&, const ProjDataInfoCylindricalNoArcCorr, const DetPairData&, const int, const int);
+
+template shared_ptr<const ProjDataInfoBlocksOnCylindricalNoArcCorr> get_fan_info_help
+(int&, int&, int&, int&, const ProjDataInfoBlocksOnCylindricalNoArcCorr& );
+template shared_ptr<const ProjDataInfoCylindricalNoArcCorr> get_fan_info_help
+(int&, int&, int&, int&, const ProjDataInfoCylindricalNoArcCorr& );
+
+template void make_fan_sum_data_help(Array<2,float>&, int, int, int, int ,
+                            const ProjDataInfoBlocksOnCylindricalNoArcCorr&, const ProjData& );
+template void make_fan_sum_data_help(Array<2,float>&, int, int,int, int,
+                            const ProjDataInfoCylindricalNoArcCorr&, const ProjData& );
+
+template void set_fan_data_add_gaps_help
+(ProjData&, int, int, int, int, const ProjDataInfoBlocksOnCylindricalNoArcCorr&, const FanProjData&, const float );
+template void set_fan_data_add_gaps_help
+(ProjData&, int, int, int, int, const ProjDataInfoCylindricalNoArcCorr&, const FanProjData&, const float );
+
+template void make_fan_data_remove_gaps_help
+(FanProjData&, int, int, int, int,  const ProjDataInfoBlocksOnCylindricalNoArcCorr&, const ProjData&);
+template void make_fan_data_remove_gaps_help
+(FanProjData&, int, int, int, int, const ProjDataInfoCylindricalNoArcCorr&, const ProjData&);
```

Comment:
These declarations serve no purpose. We'd need them if you would have used them BEFORE defining them, but you don't.
```suggestion
```